sympy/solvers/tests/test_solveset.py

```diff
@@ -2202,9 +2204,9 @@ def test_exponential_symbols():
     assert solveset(x**x, x, Interval.Lopen(0,oo)).dummy_eq(
         ConditionSet(w, Eq(w**w, 0), Interval.open(0, oo)))
     assert solveset(x**y - 1, y, S.Reals) == FiniteSet(0)
-    assert solveset(exp(x/y)*exp(-z/y) - 2, y, S.Reals) == FiniteSet(
-        (x - z)/log(2)) - FiniteSet(0)
-
+    assert solveset(exp(x/y)*exp(-z/y) - 2, y, S.Reals) == \
+    Complement(ConditionSet(y, Eq(im(x)/y, 0) & Eq(im(z)/y, 0), \
+        Complement(Intersection(FiniteSet((x - z)/log(2)), S.Reals), FiniteSet(0))), FiniteSet(0))
```

Comment:
It would be good to also add a test for x and z with real=True:
```python
In [1]: solveset(exp(x/y)*exp(-z/y) - 2, y, S.Reals)
Out[1]: 
⎧  │     ⎛    ⎧x - z ⎫⎞         ⎛im(x)       im(z)    ⎞⎫      
⎨y │ y ∊ ⎜ℝ ∩ ⎨──────⎬⎟ \ {0} ∧ ⎜───── = 0 ∧ ───── = 0⎟⎬ \ {0}
⎩  │     ⎝    ⎩log(2)⎭⎠         ⎝  y           y      ⎠⎭      

In [2]: xr, zr = symbols('xr, zr', real=True)

In [3]: solveset(exp(xr/y)*exp(-zr/y) - 2, y, S.Reals)
Out[3]: 
⎧xr - zr⎫      
⎨───────⎬ \ {0}
⎩ log(2)⎭  
```
Maybe at the top of the file add
```python
ar, br, cr, ... = symbols('ar, br, ...', real=True)
```